src/ui/pages/usersList/types.ts

```diff
@@ -0,0 +1,26 @@
+export enum UserRecipeType { 
+  emailpassword= 'emailpassword', 
+  thirdparty = "thirdparty", 
+  passwordless = "passwordless" }
+
+export type User = { recipeId: UserRecipeType.emailpassword, user: UserEmailPassword } | 
+  { recipeId: UserRecipeType.passwordless, user: UserPasswordless } |
+  { recipeId: UserRecipeType.thirdparty, user: UserThirdParty }
+
+export interface UserEmailPassword {
```

Comment:
Why do we need an interface for this?

Reply:
This to make the  `User` Type to determine automatically the correct user properties based on the `recipeId`.

Is this answering your questions?

Reply:
No I meant why use an interface and not a type?